### ISSUE TITLE
to_cartesian on differentials yields a representation not a differential

### ISSUE THREAD

**Issue by eteq** [astropy/astropy#6215]
This is primarily for @adrn and @mhvk, as it deals with #5871.

I just noticed that `<Any>Differential.to_cartesian(base)` yields a `CartesianRepresentation`, *not* a `CartesianDifferential`.  That seems very confusing.  @mhvk, is this an active design choice, and if so can you clarify why? (In which case we should update the docstring for `to_cartesian` to make it very clear.)  But if not, I can try to change it to yield differentials instead.

**Comment by mhvk** [comment]
Yes, I did choose to have `to_cartesian` mean the same thing as for representations, ie., to return a representation. This in part since for cartesian the differential and the representation are the same, and this way at least a `CartesianDifferential` can be turned into `CartesianRepresentation` (which is something that needs to be possible). I think this could be revisited, but it needs some care in `represent_as` and `from_representation`.

Looking at the docstring again, I see I tried to say that but it would probably be good to be more explicit.
```
Convert the differential to 3D rectangular cartesian coordinates.
->
Convert the differential to a 3D rectangular cartesian representation.
```

**Comment by eteq** [comment]
Gotcha.  Will issue a PR to clarify that (probably early next week since its a doc change :wink:), and will keep that this open until then.
